types/actions.pp

```diff
@@ -0,0 +1,9 @@
+# Struct data type alias for Rsyslog Actions
+type Rsyslog::Actions = Struct[{
+  String[1] => Struct[{
+    type         => Rsyslog::Action_type,
+    facility     => Optional[String[1]],
+    queue_config => Optional[Rsyslog::Queue_options],
```

Comment:
In order to implement, this will require that a breaking change PR is submitted, but this PR itself does not break any existing functionality